components-api/src/main/scala/pl/touk/nussknacker/engine/api/typed/AssignabilityDeterminer.scala

```diff
@@ -179,18 +180,50 @@ object AssignabilityDeterminer {
           )
       }
     }
-    val givenClass = from.runtimeObjType
 
-    val equalClassesOrCanAssign =
-      condNel(
-        givenClass == to,
-        (),
-        f"${givenClass.display} and ${to.display} are not the same"
-      ) orElse
-        isAssignable(givenClass.klass, to.klass)
+    // this check is done because we have GenericRecord to Map conversion in pl.touk.nussknacker.engine.spel.internal.DefaultSpelConversionsProvider
+    if (isGenericRecordToMapConversion(from, to, conversionChecker)) {
+      valid(())
+    } else {
+      val givenClass = from.runtimeObjType
 
-    val canBeSubclass = equalClassesOrCanAssign andThen (_ => typeParametersMatches(givenClass, to))
-    canBeSubclass orElse conversionChecker.isConvertable(from, to)
+      val equalClassesOrCanAssign =
+        condNel(
+          givenClass == to,
+          (),
+          f"${givenClass.display} and ${to.display} are not the same"
+        ) orElse
+          isAssignable(givenClass.klass, to.klass)
+
+      val canBeSubclass = equalClassesOrCanAssign andThen (_ => typeParametersMatches(givenClass, to))
+      canBeSubclass orElse conversionChecker.isConvertable(from, to)
+    }
+  }
+
+  private def isGenericRecordToMapConversion(
+      from: SingleTypingResult,
+      to: TypedClass,
+      conversionChecker: ConversionChecker
+  ): Boolean = {
+    if (!classOf[GenericRecord].isAssignableFrom(from.runtimeObjType.klass) || !classOf[java.util.Map[_, _]]
```

Comment:
1. We can use `IndexedRecord` which is the highest possible abstraction that fits our needs
2. Use reflection. Thanks to that we won't be statically binded with a specific avro version in api module